src/components/sidebar/LineItems.tsx

```diff
@@ -9,23 +9,16 @@ import * as s from './lineItems.scss';
 
 interface ILineItemsProps {
     lineStore?: LineStore;
-    searchInput: string;
+    location: any;
 }
 
 @inject('lineStore')
 @observer
 class LineItems extends React.Component<ILineItemsProps> {
-    componentDidMount() {
-        if (this.props.lineStore!.allLines.length === 0) {
-            LineService.getAllLines()
-                .then((lines: ILine[]) => {
-                    this.props.lineStore!.setAllLines(lines);
-                })
-                .catch((err: any) => {
-                    // tslint:disable-next-line:no-console
-                    console.log(err);
-                });
-        }
+    async componentWillMount() {
+        this.props.lineStore!.linesLoading = true;
+        await this.props.lineStore!.setAllLines(await LineService.getAllLines());
+        this.props.lineStore!.linesLoading = false;
```

Comment:
Why we have a line service call in LineItems?

And why lineStore has linesLoading? Isn't one loading variable enough in sidebarStore?

Reply:
Removed as old and unnecessary